go/hyperkit.go

```diff
@@ -0,0 +1,524 @@
+		if fn == h.DiskImage {
+			continue
+		}
+		err := os.Remove(fn)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Convert to json string
+func (h *HyperKit) String() string {
+	s, err := json.Marshal(h)
+	if err != nil {
+		return err.Error()
+	}
+	return string(s)
+}
+
+// CreateDiskImage creates a empty file suitable for use as a disk image for a hyperkit VM.
+func CreateDiskImage(location string, sizeMB int) error {
+	f, err := os.Create(location)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	buf := make([]byte, 1048676)
+	for i := 0; i < sizeMB; i++ {
+		f.Write(buf)
+	}
+	return nil
+}
+
+func (h *HyperKit) buildArgs(cmdline string) {
+	a := []string{"-A", "-u"}
+	if h.StateDir != "" {
+		a = append(a, "-F", filepath.Join(h.StateDir, pidFile))
+	}
+
+	a = append(a, "-c", fmt.Sprintf("%d", h.CPUs))
+	a = append(a, "-m", fmt.Sprintf("%dM", h.Memory))
+
+	a = append(a, "-s", "0:0,hostbridge")
+	if h.VPNKitSock != "" {
+		a = append(a, "-s", fmt.Sprintf("1:0,virtio-vpnkit,path=%s", h.VPNKitSock))
+	}
+	if h.DiskImage != "" {
+		a = append(a, "-s", fmt.Sprintf("2:0,virtio-blk,%s", h.DiskImage))
+	}
+	if h.ISOImage != "" {
+		a = append(a, "-s", fmt.Sprintf("4,ahci-cd,%s", h.ISOImage))
+	}
+
+	a = append(a, "-s", "10,virtio-rnd")
+	a = append(a, "-s", "31,lpc")
+
+	if h.Console == ConsoleFile {
+		a = append(a, "-l", fmt.Sprintf("com1,autopty=%s/tty,log=%s/console-ring", h.StateDir, h.StateDir))
+	} else {
+		a = append(a, "-l", "com1,stdio")
+	}
+
+	kernArgs := fmt.Sprintf("kexec,%s,%s,earlyprintk=serial %s", h.Kernel, h.Initrd, cmdline)
+	a = append(a, "-f", kernArgs)
+
+	h.Arguments = a
+	h.CmdLine = h.HyperKit + " " + strings.Join(a, " ")
+}
+
+// Execute hyperkit and plumb stdin/stdout/stderr.
+func (h *HyperKit) execHyperKit() error {
+
+	cmd := exec.Command(h.HyperKit, h.Arguments...)
+	cmd.Env = os.Environ()
+
+	// Plumb in stdin/stdout/stderr. If ConsoleStdio is configured
+	// plumb them to the system streams. If a logger is specified,
+	// use it for stdout/stderr logging. Otherwise use the default
+	// /dev/null.
+	if h.Console == ConsoleStdio {
+		cmd.Stdin = os.Stdin
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+	} else if h.log != nil {
+		stdoutChan := make(chan string)
+		stderrChan := make(chan string)
+		stdout, err := cmd.StdoutPipe()
+		if err != nil {
+			return err
+		}
+		stderr, err := cmd.StderrPipe()
+		if err != nil {
+			return err
+		}
+		stream(stdout, stdoutChan)
+		stream(stderr, stderrChan)
+
+		done := make(chan struct{})
+		go func() {
+			for {
+				select {
+				case stderrl := <-stderrChan:
+					log.Printf("Hyperkit ERR: ", stderrl)
+				case stdoutl := <-stdoutChan:
+					log.Printf("Hyperkit OUT: ", stdoutl)
+				case <-done:
+					return
+				}
+			}
+		}()
+	}
+
+	err := cmd.Start()
+	if err != nil {
+		return err
+	}
+	h.Pid = cmd.Process.Pid
+	h.process = cmd.Process
+	h.writeState()
+	if !h.background {
+		err = cmd.Wait()
+		if err != nil {
+			return err
+		}
```

Comment:
idomatic go would be
```
if err := cmd.Wait(); err != nil {
   return err
}
```
Here and in various other places where you don't need one of the other results from the function call, but that's down to your personal preference as author so don't feel obliged to change...